components/BottomNavigation/tests/unit/BottomNavigationTests.m

```diff
@@ -452,4 +453,52 @@ - (void)testTitlesNumberOfLinesPassedToViewsAfterItemsAssigned {
   }
 }
 
+- (void)testItemForPointInsideItemView {
+  // Given
+  UITabBarItem *item1 = [[UITabBarItem alloc] initWithTitle:@"1" image:nil tag:0];
+  UITabBarItem *item2 = [[UITabBarItem alloc] initWithTitle:@"2" image:nil tag:0];
+  self.bottomNavBar.frame = CGRectMake(0, 0, 320, 56);
+
+  // When
+  self.bottomNavBar.items = @[ item1, item2 ];
+  [self.bottomNavBar layoutIfNeeded];
+  UIView *view1 = [self.bottomNavBar viewForItem:item1];
+  UIView *view2 = [self.bottomNavBar viewForItem:item2];
+  UITabBarItem *result1 = [self.bottomNavBar tabBarItemForPoint:view1.center];
+  UITabBarItem *result2 = [self.bottomNavBar tabBarItemForPoint:view2.center];
+
+  // Then
+  XCTAssertEqualObjects(result1, item1);
+  XCTAssertEqualObjects(result2, item2);
+}
+
+- (void)testItemForPointOutsideOfItemView {
+  // Given
+  UITabBarItem *item1 = [[UITabBarItem alloc] initWithTitle:@"1" image:nil tag:0];
+  UITabBarItem *item2 = [[UITabBarItem alloc] initWithTitle:@"2" image:nil tag:0];
+  CGFloat navBarHeight = 56;
+  self.bottomNavBar.frame = CGRectMake(0, 0, 320, navBarHeight);
+  CGPoint testPoint = CGPointMake(0, navBarHeight + 10);
+
+  // When
+  self.bottomNavBar.items = @[ item1, item2 ];
+  [self.bottomNavBar layoutIfNeeded];
+  UITabBarItem *result = [self.bottomNavBar tabBarItemForPoint:testPoint];
+
+  // Then
+  XCTAssertNil(result);
+}
+
+- (void)testItemForPointNoItems {
+  // Given
+  self.bottomNavBar.frame = CGRectMake(0, 0, 320, 56);
+
+  // When
+  [self.bottomNavBar layoutIfNeeded];
+  UITabBarItem *result = [self.bottomNavBar tabBarItemForPoint:self.bottomNavBar.center];
+
+  // Then
+  XCTAssertNil(result);
+}
+
 @end
```

Comment:
Is it possible to add a test case for the case in the implementation where the items array is smaller than the item views array?

Reply:
As far as I can tell, that would be outside the scope of this PR.  It's also not clear how such a condition would arise from the public APIs.  the `itemsView` is only manipulated in `setItems:`. Do you see a code path where a caller could desynchronize the two arrays?

Since the `itemViews` property is an implementation detail, it's also not clear to me how this unit test could reasonably expect to manipulate it directly.  In that case, a whole host of errors are likely to arise and if it's necessary, a separate PR for explicitly that case should probably be produced.

Reply:
This stems primarily from the need to have a condition of `(i < self.itemViews.count) && (i < self.items.count)` in the loop above, and the implication as a result that these two values can get out of sync (I'm also not sure if/when that could be the case).

That being said, I agree that fixing this or testing for this may be out of scope for this PR if the possibility of these getting out of sync already existed prior to this PR (which I expect it does, if it does exist at all).